doc/source/database/cratedb.rst

```diff
@@ -0,0 +1,518 @@
+.. include:: ../_resources.rst
+
+.. _database-cratedb:
+
+#######
+CrateDB
+#######
+
+
+*****
+About
+*****
+
+`CrateDB`_ is a distributed and scalable SQL database for storing and analyzing massive
+amounts of data in near real-time, even with complex queries. It is PostgreSQL-compatible,
+and based on `Lucene`_.
+
+
+*******
+Details
+*******
+
+This section summarizes CrateDB's data model and query interface.
+
+Data model
+==========
+
+As a time-series/document/OLAP/RDBMS database with an SQL interface, CrateDB stores records
+into tables. Tables are grouped into schemas, which is equivalent to the concept of hosting
+multiple databases on the same server instance.
+
+The schema of tables/records can be freely defined using a classic SQL DDL statement,
+leveraging CrateDB's multi-modal data types. The tables can be queried by using classic,
+standards-compliant SQL DQL statements.
+
+.. figure:: https://github.com/daq-tools/kotori/assets/453543/fb469738-9d3e-4258-b546-1f5cd9bac261
+    :width: 640
+    :alt: A screenshot example of an SQL query submitted to CrateDB.
+
+Other than the classic record-based scheme of RDBMS databases, CrateDB also allows you
+to store and retrieve nested data, by providing `container types`_ ``ARRAY`` and ``OBJECT``.
+
+On disk, CrateDB stores data into Lucene indexes. By default, all fields are indexed,
+nested or not, but the indexing can be turned off selectively.
+
+Query interface
+===============
+
+Languages
+---------
+CrateDB supports SQL as query language. Please inspect the :ref:`cratedb-query-examples`,
+as well as the corresponding upstream documentation about how to `insert data`_ and
+`query data`_.
+
+Protocols
+---------
+CrateDB clients communicate to servers using either HTTP, or by using the `PostgreSQL
+wire protocol`_, version 3.
+
+
+************
+Key features
+************
+
+This section enumerates the key features of CrateDB, as advertised on its documentation.
+
+At a glance
+===========
+
+- Use standard SQL via the PostgreSQL wire protocol or an HTTP API.
+- Dynamic table schemas and queryable objects provide document-oriented features in
+  addition to the relational features of SQL.
+- Support for time-series data, realtime full-text search, geospatial data types and
+  search capabilities.
+- Horizontally scalable, highly available and fault tolerant clusters that run very
+  well in virtualized and containerised environments.
+- Extremely fast distributed query execution.
+- Auto-partitioning, auto-sharding, and auto-replication.
+- Self-healing and auto-rebalancing.
+- User-defined functions (UDFs) can be used to extend the functionality of CrateDB.
```

Comment:
User-defined functions are not mentioned on the canonical CrateDB README at all. It should be added.

-- https://github.com/crate/crate/blob/master/README.rst

Reply:
https://github.com/crate/crate/pull/14308